vkcs/blockstorage/resource_volume.go

```diff
@@ -214,20 +243,17 @@ func resourceBlockStorageVolumeUpdate(ctx context.Context, d *schema.ResourceDat
 
 	name := d.Get("name").(string)
 	description := d.Get("description").(string)
-	updateOpts := volumes.UpdateOpts{
+	updateOpts := VolumeUpdateOpts{
 		Name:        &name,
 		Description: &description,
-	}
-
-	if d.HasChange("metadata") {
-		metadata := d.Get("metadata").(map[string]interface{})
-		updateOpts.Metadata = util.ExpandToMapStringString(metadata)
+		Metadata:    util.ExpandToMapStringString(d.Get("metadata").(map[string]any)),
 	}
 
 	_, err = ivolumes.Update(blockStorageClient, d.Id(), updateOpts).Extract()
 	if err != nil {
 		return diag.Errorf("error updating vkcs_blockstorage_volume")
 	}
+	d.Set("metadata", updateOpts.Metadata)
```

Comment:
Let's assign configured value to a variable and use it in both places.

Reply:
done